.github/workflows/ci-cd.yml

```diff
@@ -71,18 +71,19 @@ jobs:
         run: pytest -v --markdown-docs -m markdown-docs modal
 
   container-dependencies:
-    name: Check minimal container dependencies for ${{ matrix.python-version }}
+    name: Check minimal container dependencies for ${{ matrix.python-version }} / ${{ matrix.image-builder-version }}
     runs-on: ubuntu-20.04
     timeout-minutes: 4
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        image-builder-version: ["2024.04", "2024.10"]
+        python-version: ["3.8", "3.12"]
     steps:
       - uses: actions/checkout@v3
 
       - name: Install dependencies
         run: |
-          pip install -r modal/requirements/2024.04.txt
+          pip install -r modal/requirements/${{ matrix.image-builder-version }}.txt
```

Comment:
We might want to start pinning the synchronicity version here to the version specified in the current client mount or something - this would otherwise potentially break when we release breaking changes in synchronicity without adjusting the client

Reply:
We should probably not do that :D 

(I think we should maybe just merge synchronicity into the client fwiw).